service/controller/v20/resource/node/create.go

```diff
@@ -119,6 +119,45 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted node '%s' in the tenant cluster's Kubernetes API", n.GetName()))
 	}
 
+	// Clear dead endpoint IPs only when the cluster is not in transitioning state.
+	// The amount of nodes should be equal to amount of pods.
+	if len(nodes) == len(pods) {
```

Comment:
I don't think this check is enough - this could also be coincidently the case during an update IMO.
Can you also check the status conditions somehow to see if no update is ongoing?

I am mainly just afraid of weird race conditions here.

Reply:
you don't want to wait 2 hours until KVM cluster update is finished to remove dead endpoints

we can check if node names are same as pods names but waiting for the whole update is a bad idea

Reply:
Then I don't understand the whole check at all. Why checking for any transition then?

Reply:
there can be a scenario when a new node was added to the endpoints but is not yet  in tc api ( race condition like 1-2 sec or so) and if in that time frame this will run we can have a problem that the ip would be removed

When there is the same amount of nodes and pods that means we expect all nodes in k8s TC api have a ip in the endpoint list.

Reply:
Ok but then we need to check if there is this node just added and not only compare two numbers.